### PR TITLE
Add support for wl_array type

### PR DESCRIPTION
**Pull request by zaporter**
Addresses #2 .

I have tested locally with the [wlr-foreign-toplevel-management-unstable](https://wayland.app/protocols/wlr-foreign-toplevel-management-unstable-v1). Everything is correctly typed and works as-expected.

### CODE REVIEW

File: lib/args.test.ts [hunk 1]
```diff
@@ -112,6 +120,20 @@ describe("format_args()", function() {
     expect(readFixed(buffer, 0)).to.equal(3.5);
   });
 
+  it("encodes empty array argument", function () {
+    const args = [new Uint8Array([])];
+    const def: ArgumentDefinition[] = [{ name: "arg1", type: "array" }];
+    const buffer = format_args(args, def);
+    expect(buffer.toString("hex")).to.equal("00000000");
+  });
```
Comment (zaporter) [inline lib/args.test.ts:128]: Found these empty tests helpful when debugging. The final impl does not branch on zero-size, but still useful to ensure we dont add extra bytes.

File: lib/args.ts [hunk 2]
```diff
@@ -47,13 +47,45 @@ export function readFixed(b :Buffer, offset :number, en:"LE"|"BE" = os_en) :numb
...
+  en: "LE" | "BE" = os_en,
+): number {
+  offset = writeUInt(b, array.length, offset, en);
+  b.set(array, offset);
+  const padding = (4 - (array.length % 4)) % 4;
+  if (padding > 0) {
+    b.fill(0, offset + array.length, offset + array.length + padding);
+  }
+  return offset + array.length + padding;
+}
```
Comment (zaporter) [inline lib/args.ts:81]: Implementation translated from: https://gitlab.freedesktop.org/wayland/wayland/-/blob/10df74c240d3cff4bd56bf17ebfc1a54883e00f5/src/connection.c?page=2#L1388

File: lib/args.ts [hunk 3]
```diff
@@ -47,13 +47,45 @@ export function readFixed(b :Buffer, offset :number, en:"LE"|"BE" = os_en) :numb
...
+export function readArray(
+  b: Buffer,
+  offset: number,
+  en: "LE" | "BE" = os_en,
+): [data: Uint8Array, newOffset: number] {
+  const arrayLength = readUInt(b, offset, en);
+  offset += 4;
+  const arrayData = new Uint8Array(b.subarray(offset, offset + arrayLength));
+  return [arrayData, offset + arrayLength];
+}
```
Comment (zaporter) [inline lib/args.ts:62]: Implementation translated from: https://gitlab.freedesktop.org/wayland/wayland/-/blob/10df74c240d3cff4bd56bf17ebfc1a54883e00f5/src/connection.c?page=2#L1032

**Comment by sdumetz** [comment]
Great! thanks for contributing.

**Pull request merged**